File: flutter/lib/application/use_cases/users/update_user/update_user_command_handler.dart
import 'package:enterprise_management/application/use_cases/users/update_user/update_user_command.dart';
import 'package:enterprise_management/infrastructure/repositories/user_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateUserCommandHandler extends ICommandHandler<UpdateUserCommand, bool> {
  const UpdateUserCommandHandler({required this._userRepository});
  final IUserRepository _userRepository;

  @override
  Future<Result<bool>> handle(UpdateUserCommand command) async {
    final res = await _userRepository.updateUser(id: command.id, fullName: command.fullName, email: command.email, status: command.status);
    return Result.success(res);
  }

}

File: BackEnd/Application/UseCases/Users/UpdateUser/UpdateUserCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;

namespace Application.UseCases.Users.UpdateUser;

public class UpdateUserCommandHandler(IRepository<User, Guid> userRepository, IUnitOfWork unitOfWork) : IRequestHandler<UpdateUserCommand, bool> 
{
    public async Task<bool> Handle(UpdateUserCommand request, CancellationToken cancellationToken)
    {
        var user = await userRepository.GetByIdAsync(request.Id, cancellationToken);
        if (user == null) return false;
        user.Update(request.FullName, request.Email, request.Status);
        await userRepository.UpdateAsync(user);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}